### PR TITLE
Dcm dev branch gui cont

### PR DESCRIPTION
**Pull request by adambujak**

### CODE REVIEW

File: Development/DCM/Common/callbacks.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class ApplicationCallbacks:
-    def __init__(self, loginButtonCB, logoffButtonCB, newUserButtonCB, programButtonCB, changeProgramModeCB):
+    def __init__(self, loginButtonCB, logoffButtonCB, newUserButtonCB, programButtonCB, changeProgramModeCB,createUserButtonCB,cancelButtonCB):
```
Comment (adambujak) [inline Development/DCM/Common/callbacks.py]: please fix spacing: item, item, item

File: Development/DCM/DCMDatabase/dbpm.py [hunk 2]
```diff
@@ -5,8 +5,7 @@
 
 from src.dcm_constants           import *
 from peewee                      import *
-
-from DCMUserAccountManager.duam  import * 
+import DCMUserAccountManager.duam
```
Comment (adambujak) [inline Development/DCM/DCMDatabase/dbpm.py]: why did you change this from the existing format?

File: Development/DCM/DCMDatabase/dbpm.py [hunk 3]
```diff
@@ -91,9 +90,10 @@ class BaseModel(Model):
         class Meta:
             database = database
 
-    class User(BaseModel):
+    class User1(BaseModel):
```
Comment (adambujak) [inline Development/DCM/DCMDatabase/dbpm.py]: rename to something meaningful please

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 class ScreenNames(Enum):
     LOGIN_SCREEN        = 0
     PROGRAMMING_SCREEN  = 1
+    CREATEUSER_SCREEN   = 2
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py]: RENAME to CREATE_USER_SCREEN

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 5]
```diff
@@ -43,7 +52,7 @@ def __init__(self, screenName, data):
         C_LOGIN_USERNAME_LABEL, 
         C_LOGIN_PASSWORD_LABEL, 
         C_LOGIN_BUTTON_TEXT, 
-        C_LOGIN_NEW_USER_BUTTON_TEXT))
+        C_NEW_USER_BUTTON_TEXT))
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py:62]: nice

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 6]
```diff
@@ -117,7 +140,7 @@ def p_drawLoginScreen(self, data):
 
     def p_drawProgrammingScreen(self, data):
         """
-        Draw screen of tpye programming screen
+        Draw screen of type programming screen
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py:194]: nice catch

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 7]
```diff
@@ -99,6 +118,10 @@ def getLoginData(self):
             entryData = self.gui.getEntryData()
             return LoginData(entryData[0], entryData[1])
 
+    def getNewUserData(self):
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py:153]: add function brief by using triple quotes

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 8]
```diff
@@ -126,8 +149,15 @@ def p_drawProgrammingScreen(self, data):
             data.fieldLabels,
             data.buttonTexts,
             data.buttonCallbacks)
+
+    def p_drawCreateUserScreen(self, data):  #create User and cancel
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py]: function brief

File: Development/DCM/DCMGraphicalUserInterface/readme.txt [hunk 9]
```diff
@@ -1,2 +1,5 @@
 __init__.py is to indidcate that directory shall be used as import directory
-guic.py is the Graphical User Interface Controller source file
\ No newline at end of file
+guic.py is the Graphical User Interface Controller source file
+guial.py is the Graphical User Interface Abstraction Layer Class
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/readme.txt:3]: thanks for updating

File: Development/DCM/DCMUserAccountManager/duam.py [hunk 10]
```diff
@@ -87,12 +87,14 @@ def signInUser(self, p_loginData):
             passwordValid = verify_password(userData.getPassword(), password)
             # Redundant check to make sure correct user data is being returned
             usernameValid = (userData.getUsername() == username)
-
+            print(userData.getUsername() + userData.getPassword())
+            print(passwordValid)
```
Comment (adambujak) [inline Development/DCM/DCMUserAccountManager/duam.py]: what's the point of these? if debugging, it's a good idea to put a label on it - print("passwordValid:", passwordValid)

File: Development/DCM/main.py [hunk 11]
```diff
@@ -25,19 +25,27 @@ def updateGUI(self):
...
 
     def logoffButtonCB(self):
         self.guiController.drawScreen(loginScreen)
 
     def newUserButtonCB(self):
-        print("login button callback")
+        self.guiController.drawScreen(createUserScreen)
 
+    def createUserButtonCB(self):
+        x = self.accountController.makeNewUser(self.guiController.getNewUserData(),"hello")
```
Comment (adambujak) [inline Development/DCM/main.py]: is hello the admin password? if so, please change it to "adminPassword" so that it's not just a random hello in the code. Probably best to just add it as a constant

File: Development/DCM/main.py [hunk 12]
```diff
@@ -48,9 +56,16 @@ def logoffButtonCallback():
...
+
+    def createUserButtonCallback():
+        app.createUserButtonCB()
+
+    def cancelButtonCallback():
+        app.cancelButtonCB()
+
 
-    callbacks = ApplicationCallbacks(loginButtonCallback, logoffButtonCallback, newUserButtonCallback, None, None)
+    callbacks = ApplicationCallbacks(loginButtonCallback, logoffButtonCallback, newUserButtonCallback, None, None,createUserButtonCallback,cancelButtonCallback)
```
Comment (adambujak) [inline Development/DCM/main.py]: spacing

File: Development/DCM/src/dcm_constants.py [hunk 13]
```diff
@@ -30,8 +30,8 @@
 
 C_LOGIN_USERNAME_LABEL         =       "Username"                                         # LABEL VALUE FOR LOGIN USERNAME FIELD 
 C_LOGIN_PASSWORD_LABEL         =       "Password"                                         # LABEL VALUE FOR LOGIN PASSWORD FIELD 
-C_LOGIN_BUTTON_TEXT            =       "Sign In"                                          # BUTTON TEXT VALUE FOR LOGIN BUTTON
-C_LOGIN_NEW_USER_BUTTON_TEXT   =       "New User"                                         # BUTTON TEXT VALUE FOR NEW USER BUTTON
+C_LOGIN_BUTTON_TEXT            =       "Login"                                          # BUTTON TEXT VALUE FOR LOGIN BUTTON
+C_NEW_USER_BUTTON_TEXT   	   =       "New User?"                                         # BUTTON TEXT VALUE FOR NEW USER BUTTON
```
Comment (adambujak) [inline Development/DCM/src/dcm_constants.py]: fix alignment of comments please

File: Development/DCM/src/dcm_constants.py [hunk 14]
```diff
@@ -40,4 +40,7 @@
 C_PROGRAM_UPPER_LIMIT_LABEL    =       "Upper Rate Limit"                                 # LABEL VALUE FOR UPPER RATE LIMIT FIELD 
 C_PROGRAM_LOWER_LIMIT_LABEL    =       "Lower Rate Limit"                                 # LABEL VALUE FOR LOWER RATE LIMIT FIELD 
 C_PROGRAM_BUTTON_TEXT          =       "Program"                                          # BUTTON TEXT VALUE FOR PROGRAM BUTTON
-C_PROGRAM_LOGOUT_BUTTON_TEXT   =       "Sign Out"                                         # BUTTON TEXT VALUE FOR LOGOUT BUTTON
\ No newline at end of file
+C_PROGRAM_LOGOUT_BUTTON_TEXT   =       "Sign Out"                                         # BUTTON TEXT VALUE FOR LOGOUT BUTTON
+
+C_CREATE_USER_BUTTON_TEXT	   = 	   "Create New User"
```
Comment (adambujak) [inline Development/DCM/src/dcm_constants.py]: comments

File: Development/DCM/DCMDatabase/dbpm.py (hunk 2 above)
Comment (EAHillebrand) [reply Development/DCM/DCMDatabase/dbpm.py]: Still attempting to get the class construction on line 62 to work, which use the class User from DCMUserAccountManager.duam

File: Development/DCM/src/dcm_constants.py (hunk 14 above)
Comment (EAHillebrand) [reply Development/DCM/src/dcm_constants.py]: I dont see this issue in my text editor

File: Development/DCM/DCMDatabase/dbpm.py (hunk 2 above)
Comment (adambujak) [reply Development/DCM/DCMDatabase/dbpm.py]: those two lines are equivalent though?

File: Development/DCM/DCMGraphicalUserInterface/guic.py [hunk 15]
```diff
@@ -119,7 +119,10 @@ def getLoginData(self):
             return LoginData(entryData[0], entryData[1])
 
     def getNewUserData(self):
-        if self.currentScreen.screenName == ScreenNames.CREATEUSER_SCREEN:
+        '''
```
Comment (adambujak) [inline Development/DCM/DCMGraphicalUserInterface/guic.py]: Either use double quotes everywhere, or single quotes everywhere please

File: Development/DCM/DCMDatabase/dbpm.py [hunk 16]
```diff
@@ -5,15 +5,26 @@
...
 database = SqliteDatabase(C_DATABASE_PATH, pragmas={
     'journal_mode': C_DATABASE_JOURNAL_MODE,
     'cache_size': C_DATABASE_CACHE_SIZE,
     'foreign_keys': C_DATABASE_FOREIGN_KEYS})
 
-
+class User:
+    def __init__(self, p_username, p_password, p_userRole):
+        self.username    = p_username
+        self.password    = p_password
```
Comment (adambujak) [inline Development/DCM/DCMDatabase/dbpm.py:20]: this shouldn't be here.. this doesn't relate to the database user data, this relates to the user in the user account manager

File: Development/DCM/main.py [hunk 17]
```diff
@@ -25,27 +25,35 @@ def updateGUI(self):
...
+        
+
 
     def cancelButtonCB(self):
         self.guiController.drawScreen(loginScreen)
 
 
 def main():
-    #print(hash_password('hello')) #Displays hashed Admin password
+    #print(hash_password("C_ADMIN_PASSWORD")) #Displays hashed C_ADMIN_PASSWORD
```
Comment (adambujak) [inline Development/DCM/main.py:85]: C_ADMIN_PASSWORD may be confusing since it follows the same scheme as our constant variable names
